File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task57_PERPLEXITY_llama-3-sonar-large-32k-chat.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter a number: ");
        int num = scanner.nextInt();
        long factorial = 1;
        for(int i = 1; i <= num; i++) {
            factorial *= i;
        }
        System.out.println("Factorial of " + num + " is " + factorial);
    }
}

File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task57_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
#include<iostream>
using namespace std;

long long factorial(int n) {
    long long fact = 1;
    for(int i = 1; i <= n; i++) {
        fact *= i;
    }
    return fact;
}

int main() {
    int num;
    cout << "Enter a number: ";
    cin >> num;
    cout << "The factorial of " << num << " is " << factorial(num) << endl;
    return 0;
}